### PR TITLE
Digicarp65

### PR DESCRIPTION
**Pull request by swiftrax**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/swiftrax/digicarp65/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    ),
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise)
+        tap_code16(KC_VOLU);
+    else
+        tap_code16(KC_VOLD);
+    return true;
+}
```
Comment (drashna) [inline keyboards/handwired/swiftrax/digicarp65/keymaps/via/keymap.c:57]:
```suggestion
```

File: keyboards/handwired/swiftrax/digicarp65/digicarp65.c [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "digicarp65.h"
```
Comment (drashna) [inline keyboards/handwired/swiftrax/digicarp65/digicarp65.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "digicarp65.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/swiftrax/digicarp65/readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# DigiCarp65
+
+A 65% keyboard
+
+* Keyboard Maintainer: Swiftrax
+* Hardware Supported: DigiCarp65
+* Hardware Availability: https://github.com/swiftrax
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/handwired/swiftrax/digicarp65/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/swiftrax/digicarp65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes        # Rotary Encder
```
Comment (fauxpark) [inline keyboards/handwired/swiftrax/digicarp65/rules.mk:19]:
```suggestion
ENCODER_ENABLE = yes        # Rotary Encoder

LAYOUTS = 65_ansi 65_ansi_split_bs 65_iso 65_iso_split_bs
```

?

File: keyboards/handwired/swiftrax/digicarp65/digicarp65.h [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+     K40, K41, K42,                K46,                K4A, K4B, K4C, K4D, K4E, K4F  \
+ ) \
+ { \
+     { K00, K01,   K02,   K03,   K04,   K05, K06,   K07,   K08, K09,   K0A, K0B,   K0C,   KC_NO,   K0E, K0F }, \
+     { K10, KC_NO, K12,   K13,   K14,   K15, K16,   K17,   K18, K19,   K1A, K1B,   K1C,     K1D,   K2D, K1F }, \
+     { K20, KC_NO, K22,   K23,   K24,   K25, K26,   K27,   K28, K29,   K2A, K2B,   K2C,   KC_NO,   K2E, K2F }, \
+     { K30,   K31, K32,   K33,   K34,   K35, K36,   K37,   K38, K39,   K3A, K3B, KC_NO,     K3D,   K3E, K3F }, \
+     { K40, K41,   K42, KC_NO, KC_NO, KC_NO, K46, KC_NO, KC_NO, KC_NO, K4A, K4B,   K4C,     K4D,   K4E, K4F }, \
+ }
+ #define LAYOUT_65_ansi_splitbs( \
```
Comment (fauxpark) [inline keyboards/handwired/swiftrax/digicarp65/digicarp65.h:51]:
```suggestion
 #define LAYOUT_65_ansi_split_bs( \
```

File: keyboards/handwired/swiftrax/digicarp65/digicarp65.h [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+     K40, K41, K42,                K46,                K4A, K4B, K4C, K4D, K4E, K4F  \
+ ) \
+ { \
+     { K00, K01,   K02,   K03, K04,   K05,   K06, K07,   K08,   K09,   K0A, K0B, K0C,   K0D,   K0E, K0F }, \
+     { K10, KC_NO, K12,   K13, K14,   K15,   K16, K17,   K18,   K19,   K1A, K1B, K1C,   K1D,   K1E, K1F }, \
+     { K20, KC_NO, K22,   K23, K24,   K25,   K26, K27,   K28,   K29,   K2A, K2B, K2C,   KC_NO, K2E, K2F }, \
+     { K30, KC_NO, K32,   K33, K34,   K35,   K36, K37,   K38,   K39,   K3A, K3B, KC_NO, K3D,   K3E, K3F }, \
+     { K40, K41,   K42, KC_NO, KC_NO, KC_NO, K46, KC_NO, KC_NO, KC_NO, K4A, K4B,   K4C, K4D,   K4E, K4F }, \
+ }
+#define LAYOUT_65_iso_splitbs( \
```
Comment (fauxpark) [inline keyboards/handwired/swiftrax/digicarp65/digicarp65.h:65]:
```suggestion
#define LAYOUT_65_iso_split_bs( \
```

File: keyboards/handwired/swiftrax/digicarp65/info.json [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+                {"label": "r5_03", "x": 2.50, "y": 4, "w": 1.25},
+                {"label": "r5_04", "x": 3.75, "y": 4, "w": 6.25},
+                {"label": "r5_05", "x": 10, "y": 4, "w": 1.25},
+                {"label": "r5_06", "x": 11.25, "y": 4, "w": 1.25},
+                {"label": "r4_07", "x": 13, "y": 4},
+                {"label": "r4_08", "x": 14, "y": 4},
+                {"label": "r4_09", "x": 15, "y": 4}
+            ]
+        },
+        "LAYOUT_65_ansi_splitbs": {
```
Comment (fauxpark) [inline keyboards/handwired/swiftrax/digicarp65/info.json:159]:
```suggestion
        "LAYOUT_65_ansi_split_bs": {
```

File: keyboards/handwired/swiftrax/digicarp65/info.json [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+                {"label": "r5_03", "x": 2.50, "y": 4, "w": 1.25},
+                {"label": "r5_04", "x": 3.75, "y": 4, "w": 6.25},
+                {"label": "r5_05", "x": 10, "y": 4, "w": 1.25},
+                {"label": "r5_06", "x": 11.25, "y": 4, "w": 1.25},
+                {"label": "r4_07", "x": 13, "y": 4},
+                {"label": "r4_08", "x": 14, "y": 4},
+                {"label": "r4_09", "x": 15, "y": 4}
+            ]
+        },
+        "LAYOUT_65_iso_splitbs": {
```
Comment (fauxpark) [inline keyboards/handwired/swiftrax/digicarp65/info.json:235]:
```suggestion
        "LAYOUT_65_iso_split_bs": {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/swiftrax/digicarp65/rules.mk (hunk 4 above)
Comment (swiftrax) [reply keyboards/handwired/swiftrax/digicarp65/rules.mk]: probably accidentally deleted

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
> ☒ handwired/swiftrax/digicarp65: Claims to support a community layout that does not exist: 65_iso_split_bs

**Comment by fauxpark** [comment]
I can whip up a default layout for `65_iso_split_bs` in a bit.

**Comment by fauxpark** [comment]
@swiftrax it seems like the LAYOUT macros and info.json don't match up here. Does this board support the normal `65_ansi` layout *and* `65_ansi_blocker`, or just one of them?

**Comment by swiftrax** [comment]
> @swiftrax it seems like the LAYOUT macros and info.json don't match up here. Does this board support the normal `65_ansi` layout _and_ `65_ansi_blocker`, or just one of them?

It only supports `65_ansi` not blocker

**Pull request merged**